Sources/ArcGISToolkit/Components/AR/ARSwiftUIView.swift

```diff
@@ -0,0 +1,165 @@
+// Copyright 2023 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ARKit
+import SwiftUI
+
+typealias ARViewType = ARSCNView
+
+/// A SwiftUI version of an AR view.
+struct ARSwiftUIView {
+    /// The closure to call when the AR view renders.
+    private(set) var onDidUpdateFrameAction: ((ARSession, ARFrame) -> Void)?
+    private(set) var onAddNodeAction: ((SCNSceneRenderer, SCNNode, ARAnchor) -> Void)?
+    private(set) var onUpdateNodeAction: ((SCNSceneRenderer, SCNNode, ARAnchor) -> Void)?
+    private(set) var onSingleTapGesture: ((CGPoint) -> Void)?
+    private(set) var videoFeedIsHidden: Bool = false
+    
+    /// The proxy.
+    private let proxy: ARSwiftUIViewProxy?
+    
+    /// Creates an ARSwiftUIView.
+    /// - Parameter proxy: The provided proxy which will have it's state filled out
+    /// when available by the underlying view.
+    init(proxy: ARSwiftUIViewProxy? = nil) {
+        self.proxy = proxy
+    }
+    
+    /// Sets the closure to call when underlying scene renders.
+    func onDidUpdateFrame(
+        perform action: @escaping (ARSession, ARFrame) -> Void
+    ) -> Self {
+        var view = self
+        view.onDidUpdateFrameAction = action
+        return view
+    }
+    
+    /// Sets the closure to call when a new node has been added to the scene.
+    func onAddNode(
+        perform action: @escaping (SCNSceneRenderer, SCNNode, ARAnchor) -> Void
+    ) -> Self {
+        var view = self
+        view.onAddNodeAction = action
+        return view
+    }
+    
+    /// Sets the closure to call when the scene's nodes are update.
+    func onUpdateNode(
+        perform action: @escaping (SCNSceneRenderer, SCNNode, ARAnchor) -> Void
+    ) -> Self {
+        var view = self
+        view.onUpdateNodeAction = action
+        return view
+    }
+    
+    /// Sets a closure to perform when a single tap occurs on the AR view.
+    func onSingleTapGesture(
+        perform action: @escaping (CGPoint) -> Void
+    ) -> Self {
+        var view = self
+        view.onSingleTapGesture = action
+        return view
+    }
```

Comment:
This is fine, but it's not ideal because we won't be reminded to remove it when we no longer support iOS 16. I would prefer a general `View` extension like the presentation detents one.